pkg/transformer/openshift/openshift.go

```diff
@@ -245,9 +243,19 @@ func initBuildConfig(name string, service kobject.ServiceConfig, repo string, br
 
 // initDeploymentConfig initialize OpenShifts DeploymentConfig object
 func (o *OpenShift) initDeploymentConfig(name string, service kobject.ServiceConfig, replicas int) *deployapi.DeploymentConfig {
-	tag := getImageTag(service.Image)
 	containerName := []string{name}
 
+	// Use "name" by default (if service.Image isn't set)
+	image := name
+	// Use image if it was set
+	if service.Image != "" {
+		image = service.Image
+	}
+	// Add a tag if not present
+	if !strings.Contains(image, ":") {
+		image = image + ":" + getImageTag(service.Image)
+	}
+
```

Comment:
Since this is getting used at multiple places, how about extracting this out to a util function? If we don't want the tag to be set (like in BuildConfig), we can tweak that in the function definition, WDYT?